pkgs/desktops/gnome/core/gnome-keyring/default.nix

```diff
@@ -72,6 +75,7 @@ stdenv.mkDerivation rec {
 
   postPatch = ''
     patchShebangs build
+    addToSearchPath ACLOCAL_PATH ${lib.getDev libgcrypt}/share/aclocal
```

Comment:
Why is this not handled by the automake setup hook?

Reply:
if i understand it right, `libgcrypt` includes m4 macros in its `dev` output which are used during build time. the `automake` hook populates `ACLOCAL_PATH` for this kind of thing, but [only for packages in `nativeBuildInputs`](https://github.com/NixOS/nixpkgs/blob/bfb7a882678e518398ce9a31a881538679f6f092/pkgs/development/tools/misc/automake/setup-hook.sh#L5) (slight simplification).

other ways to fix this is:
1. place `libgcrypt` in both `buildInputs` and `nativeBuildInputs`.
2. tweak that automake setup-hook to populate ACLOCAL_PATH also from the `$targetOffset` (?) in addition to `$hostOffset`, and keep libgcrypt in just `buildInputs`.
3. build gnome-keyring in such a way that it doesn't need libgcrypt's m4 file.

i suspect most users are doing option 1. it's pretty wasteful in terms of compute/builder resources, but i suppose it's more maintainable than manually setting ACLOCAL_PATH here.

Reply:
oh, or we can just avoid this altogether by not invoking autoconf and using their vendored `configure` instead 🧠